x-pack/plugins/apm/public/components/shared/ImpactBar/index.tsx

```diff
@@ -8,13 +8,61 @@ import { EuiProgress } from '@elastic/eui';
 import React from 'react';
 import { StringMap } from '../../../../typings/common';
 
+export interface ImpactItem extends StringMap<any> {
+  impact: number;
+}
+
 // TODO: extend from EUI's EuiProgress prop interface
-interface Props extends StringMap<any> {
+export interface ImpactBarProps extends StringMap<any> {
   value: number;
+  max?: number;
+}
+
+export interface RelativeImpactGroupRenderProps extends StringMap<any> {
+  relativeImpactItems: ImpactItem[];
+  RelativeImpactBar: (props: ImpactBarProps) => JSX.Element;
+  impactMeta: {
+    baseline: number;
+    max: number;
+    relativeMax: number;
+  };
+}
+
+export function RelativeImpactGroup({
+  items,
+  render,
+  impactKey = 'impact'
+}: {
+  items: ImpactItem[];
+  render: (props: RelativeImpactGroupRenderProps) => any;
+  impactKey?: string;
+}) {
+  const values = items.map(item => item[impactKey]);
+  const baseline = Math.min(...values);
+  const max = Math.max(...values);
+  const relativeMax = max - baseline;
+
+  const calculated = items.map(item => ({
+    ...item,
+    impact: item.impact - baseline,
+    rawImpact: item.impact
+  }));
+
+  return render({
+    relativeImpactItems: calculated,
+    RelativeImpactBar: (props: ImpactBarProps) => (
+      <ImpactBar max={relativeMax} {...props} />
+    ),
+    impactMeta: {
+      baseline,
+      max,
+      relativeMax
+    }
+  });
 }
```

Comment:
I think this component seem a overly complicated. Why not just calculated it in Node, and just use `EuiProgress` directly? Then we can also avoid the render prop. I prefer render props over HOC but but they both add complexity, and I don't think it's necessary in this case.

Reply:
If we calculate it in node then we have to pass along the "max" value somehow. I didn't see a simple and clean way of doing that without having to calcuate it multiple times.

Reply:
Do we need the max value? Can't we just calculate the relative impact as percentage, and provide that to `EuiProgress`? Implicitly the max will be 100 but we don't have to pass that along.